File: backend/Stoctable.Domain/Entities/Sale.cs
using Stoctable.Domain.Entities.Base;
using Stoctable.Domain.Enums;

namespace Stoctable.Domain.Entities;

public class Sale : BaseEntity
{
    public string SaleNumber { get; set; } = string.Empty;
    public Guid? QuotationId { get; set; }
    public Guid? CustomerId { get; set; }
    public Guid? SalespersonId { get; set; }
    public Guid? CashierId { get; set; }
    public SaleStatus Status { get; set; } = SaleStatus.PendingPayment;
    public decimal Subtotal { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal AmountPaid { get; set; }
    public string? Notes { get; set; }
    public DateTimeOffset? CompletedAt { get; set; }

    public Customer? Customer { get; set; }
    public User? Salesperson { get; set; }
    public User? Cashier { get; set; }
    public ICollection<SaleItem> Items { get; set; } = [];
    public ICollection<Payment> Payments { get; set; } = [];
}

File: frontend/src/types/sale.ts
export type SaleStatus = 'pending_payment' | 'partially_paid' | 'paid' | 'cancelled';
export type PaymentStatus = 'pending' | 'completed' | 'refunded';

export interface SaleItem {
  id: string;
  productId: string;
  productSku: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  discountPct: number;
  lineTotal: number;
}

export interface Payment {
  id: string;
  saleId: string;
  paymentMethodId: string;
  paymentMethodName: string;
  amount: number;
  installments: number;
  status: PaymentStatus;
  transactionRef?: string;
  paidAt?: string;
}

export interface PaymentMethod {
  id: string;
  name: string;
  requiresInstallments: boolean;
  maxInstallments: number;
  isActive: boolean;
}

export interface Sale {
  id: string;
  saleNumber: string;
  quotationId?: string;
  customerId?: string;
  customerName?: string;
  salespersonId: string;
  salespersonName: string;
  cashierId?: string;
  cashierName?: string;
  status: SaleStatus;
  subtotal: number;
  discountAmount: number;
  totalAmount: number;
  amountPaid: number;
  notes?: string;
  completedAt?: string;
  items: SaleItem[];
  payments: Payment[];
  createdAt: string;
}

export interface ProcessPaymentRequest {
  payments: {
    paymentMethodId: string;
    amount: number;
    installments?: number;
    transactionRef?: string;
  }[];
}
